werr/errors.go

```diff
@@ -0,0 +1,35 @@
+package werr
+
+import (
+	"context"
+	"fmt"
+)
+
+// WrappedError represents a standard error
+// that can also encapsulate a context.
+type WrappedError struct {
+	Ctx context.Context
+	Err error
```

Comment:
I don't think we need to include an `error` field in this struct, just the context. The wrapping of errors should follow that standard golang mechanism.
To add a context to an error you'd write:
```go
...
return errors.Wrap(err, werr.WrappedError(ctx))
```

Reply:
But then how would I `unwrap` the WrappedError? The reason that i added the `Err` field in this was to be able to unwrap the error and get back the original error.

```
func (w *WrappedError) Unwrap() error {
	return w.Err
}
```